src/drivers/ports/esp32/can_port.i

```diff
@@ -269,31 +363,33 @@ int can_port_start(struct can_driver_t *self_p)
 
     regs_p->OCTRL = (ESP32_CAN_OCTRL_MODE_NORMAL);
 
-    /* Clear error counters and error code capture */
-    regs_p->TXERR = 0;
-    regs_p->RXERR = 0;
-    (void)regs_p->ECC;
-
-    /* Clear interrupt flags by reading the interrupt status
-       register.  */
-    (void)regs_p->INT;
-
-    /* Install the interrupt handler. */
-    esp_xt_set_interrupt_handler(ESP32_CPU_INTR_CAN_NUM,
-                                 isr,
-                                 NULL);
-    esp_xt_ints_on(BIT(ESP32_CPU_INTR_CAN_NUM));
-
-    /* Map the CAN peripheral interrupt to the CAN CPU interrupt. */
-    intr_matrix_set(xPortGetCoreID(),
-                    ESP32_INTR_SOURCE_CAN,
-                    ESP32_CPU_INTR_CAN_NUM);
-
-    /* Set chip to normal mode. */
-    regs_p->MODE = 0;
+    reset_hw(self_p);
 
     dev_p->drv_p = self_p;
 
+    /* Check CAN-RX pin is connected to the trasceiver.
+       Return error if "CAN-module is waiting to be idle again".
+       CAN-module in this state is waiting for 11 consecutive recessive bits,
+       so we will wait for at least 170 bit-periods (128*1.2+11 = 164, where
+       128 - is maximum CAN-frame length, 1.2 - bit stuffing koeff). */
+    still_waiting_for_bus_idle = 1;
+    time_get(&now);
+    delta.seconds = 0;
+    delta.nanoseconds = 17000L * 1000;    // 17000 = 170 * 100: 170 cycles of bit-period @ 10 KBPS (100 us per bit)
+    time_add(&stop, &now, &delta);
+    while ( (now.seconds < stop.seconds) || (now.nanoseconds < stop.nanoseconds) ) {   
+
+        still_waiting_for_bus_idle = is_waiting_for_bus_idle(self_p);
+        if ( ! still_waiting_for_bus_idle ) {
+            break;
+        }    
+
+        time_get(&now);
+    }
+    if ( still_waiting_for_bus_idle  ) {
+        return (-ENETDOWN);
+    }
+    
```

Comment:
Put this code in a function called ``int wait_for_bus_idle()``, and return 0 from inside the while loop if the bus becomes idle, and returns -ENETDOWN at the end of the function when a timeout occurred.